### PR TITLE
Add lineNumberProps and lineNumberContainerProps (breaking)

### PR DESCRIPTION
**Pull request by masterzenox**
This change allows for greater customizability and is more in line with
`lineProps`.

### CODE REVIEW

**Comment by Roeefl** [comment]
Nice!

**Comment by conorhastings** [comment]
@masterzenox does this break any default behavior or is it a breaking change to a way additive behavior works

**Comment by masterzenox** [comment]
It removes `lineNumberStyle` and `lineNumberContainerStyle` in favour of having a props object, so
```jsx
<SyntaxHighlighter lineNumberStyle={{ cursor: "pointer" }}>
  {someCode}
</SyntaxHighlighter>
```
would have to be refactored into
```jsx
<SyntaxHighlighter 
  lineNumberProps={{ 
    style: { cursor: "pointer" }
  }}
>
  {someCode}
</SyntaxHighlighter>
```

File: src/highlight.js [hunk 1]
```diff
@@ -24,15 +27,17 @@ function getLineNumbers({ lines, startingLineNumber, style }) {
 function LineNumbers({
   codeString,
   codeStyle,
-  containerStyle = { float: 'left', paddingRight: '10px' },
-  numberStyle = {},
+  containerProps = { style: { float: 'left', paddingRight: '10px' } },
+  numberProps,
```
Comment (conorhastings) [inline src/highlight.js]: I have an idea to make this a non breaking change will pr your pr

**Comment by conorhastings** [comment]
@masterzenox actually decided i prefrer just to make breaking change can you update readme and add add tests by any chacne?

**Comment by masterzenox** [comment]
@conorhastings sure

**Comment by conorhastings** [comment]
thanks will get in quickly when that's complete

**Comment by conorhastings** [comment]
had a couple issues, this should be in as 12.0.2

**Comment by masterzenox** [comment]
@conorhastings it seems you only bumped up the version without commiting the actual changes

**Comment by conorhastings** [comment]
@masterzenox can you try 12.2.1 sorry for the verrsion thrash having build issues on updated laptop

**Comment by masterzenox** [comment]
@conorhastings I'm on mobile, so maybe it's just on my end, but the tag 12.2.1 isn't merged with master (but the demo works)?

**Comment by conorhastings** [comment]
may have neglected to push tag will check at work

**Comment by SalahAdDin** [comment]
Bug here:
```
react_devtools_backend.js:562 Warning: React does not recognize the `lineNumberStyle` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `linenumberstyle` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in pre (created by SyntaxHighlighter)
    in SyntaxHighlighter (at content.js:14)
    in CodeBlock (created by ReactMarkdown)
    in Root (created by ReactMarkdown)
    in ReactMarkdown (at content.js:88)
    in Content (at personalinformation.js:87)
    in div (created by ForwardRef(Typography))
    in ForwardRef(Typography) (created by WithStyles(ForwardRef(Typography)))
    in WithStyles(ForwardRef(Typography)) (at personalinformation.js:86)
    in div (created by ForwardRef(CardContent))
    in ForwardRef(CardContent) (created by WithStyles(ForwardRef(CardContent)))
    in WithStyles(ForwardRef(CardContent)) (at personalinformation.js:84)
    in section (created by ForwardRef(Paper))
    in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
    in WithStyles(ForwardRef(Paper)) (created by ForwardRef(Card))
    in ForwardRef(Card) (created by WithStyles(ForwardRef(Card)))
    in WithStyles(ForwardRef(Card)) (at personalinformation.js:38)
    in PersonalInformation (at about-me.js:80)
    in main (created by ForwardRef(Container))
    in ForwardRef(Container) (created by WithStyles(ForwardRef(Container)))
    in WithStyles(ForwardRef(Container)) (at layout.js:26)
    in Layout (at about-me.js:38)
    in AboutMe (at _app.js:28)
    in App
    in ErrorBoundary (created by ReactDevOverlay)
    in ReactDevOverlay (created by Container)
    in Container (created by AppContainer)
    in AppContainer
    in Root
```

**Pull request merged**